### DISCUSSION TITLE
How do I play a sound?

### DISCUSSION THREAD

**Discussion by ufomike** [jef/streetmerchant#1019] [Help]
### Description

Do I just paste the path to an MP3 file on my hard disk in the PLAY_SOUND, section to get it the sound to work on the .env files. example: PLAY_SOUND=C:\Users\Michael Rig\streetmerchant\beep.MP3

**Comment by ufomike** [comment]
I tried to insert a path to an mp3 files but got this error message: INFO: Could not find files for the given pattern(s)

**Comment by huntinganswers** [comment]
i have the same error message

**Comment by mukul12** [comment]
Try inserting that in quotes: "C:\Users\Michael Rig\streetmerchant\beep.MP3"

Also try not to have Space in path e.g Michael Rig -> Michael_Rig 

If not Try this as well: "C:\\Users\\Michael Rig\\streetmerchant\\beep.MP3"
The second time I have 2 \ \ without space in between

**Comment by ufomike** [comment]
@mukul12 No, none of that works. I already tried with and without quotes. As far as the space in my rig name that is the correct spelling of my computer name, spaces and all. The problem from what I can tell so is, even though the path to the file is correct the mp3 files is not executing as a player needs to be invoked.  Looking into writing a script in PowerShell. As a batch file did not work. Maybe I'm making this harder than it is but without an alarm this script npm script is useless.

**Comment by huntinganswers** [comment]
im in the same boat ufomike

**Comment by kwsim539** [comment]
I am having the same issue. I have the .mp3 file in the root of the streetmerchant dir and the setting set to PLAY_SOUND="C:\streetmerchant\quite-impressed-565.mp3". I am running Node 14.15.1. When I run the test:notification, the toast notification works fine but getting the following error for the sound and it can't find the sound player.

INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
[7:26:31 PM] warn :: ✖ couldn't find sound player
[7:26:32 PM] info :: ✔ desktop notification sent

**Comment by huntinganswers** [comment]
thats interesting kswim.  I get the same pattern warning, but says my sound player is found.  

The test notification plays the sound and I get the notification email, but the given pattern(s) notification continues.

**Comment by kwsim539** [comment]
> thats interesting kswim. I get the same pattern warning, but says my sound player is found.
> 
> The test notification plays the sound and I get the notification email, but the given pattern(s) notification continues.

Where is your build located? This might have something to do with the finding the sound player error.

**Comment by huntinganswers** [comment]
![error](https://user-images.githubusercontent.com/75166450/100563061-1a671580-3283-11eb-96dc-5429610475e7.JPG)

**Comment by kwsim539** [comment]
@huntinganswers Hmm, looks like your build is under your user dir. Mine is in the C:\ root. This might be the issue. Let me rebuild this in my user dir and give it another try. Thnx.

**Comment by kwsim539** [comment]
@huntinganswers Same issue:

![image](https://user-images.githubusercontent.com/8410150/100563849-0fad8000-3285-11eb-9d93-20e651f4c2bb.png)

**Comment by huntinganswers** [comment]
thats just weird!  Im not using powershell either.  This might be a stupid question ... do you have all 4 of the mp3 player files in the streetmerchant folder?
![player](https://user-images.githubusercontent.com/75166450/100564100-a5e1a600-3285-11eb-8fb7-a9644af2528b.JPG)

**Comment by kwsim539** [comment]
I do not have these. They are not part of the repo and was not placed during the install.

![image](https://user-images.githubusercontent.com/8410150/100564280-36b88180-3286-11eb-971b-b8bf87a16f91.png)

**Comment by huntinganswers** [comment]
unzip these to the root streetmerchant folder
http://www.mailsend-online.com/wp/cmdmp3new.zip

**Comment by kwsim539** [comment]
@huntinganswers Yep that worked. Thnx. Now just need to work out the "Could not find files for the given pattern(s)" issue. It seems to be working besides that.

**Comment by huntinganswers** [comment]
nice.

I think perhaps the command or whatever is simply looking for the player earlier in the chain, but finds it after..if that makes sense?  I'm not a coding expert - only ok with attempting to understand.

**Comment by ufomike** [comment]
In my case the script can't find my mp3 file or the player so created a batch file that open a PowerShell window and invokes a player and plays the mp3 file. then populated the .env file where it says PLAY_SOUND. 
The syntax for the batch file goes like this: start PowerShell Invoke-Item "C:\music\alarm.mp3"
I created and saved the batch file. Before placing the batch file into the .env file I tested the file location of the batch file in the run command in windows10 and it run just fine. The exact file location for this batch file is C:\Users\Michael Rig\Desktop\mp3.bat.   It did what it was supposed to do. Invoked a PowerShell window run the player and played the MP3 file.
I entered the exact file location of the batch file into the .env file in the PLAY_SOUND= "C:\Users\Michael Rig\Desktop\mp3.bat" saved the .env file and run the scrip again. No luck, it did not work.

**Comment by jef** [comment]
I would put the music file where you've put streetmerchant and reference it like `PLAY_SOUND=alarm.mp3`. This is the easiest method.

Also take a look at this: https://github.com/jef/streetmerchant/wiki/Installation:-Windows#audio-notifications

**Comment by gtoravan** [comment]
I am having similar issue. After extracting all cmdmp3 files to streetmerchant, I am not able to open player. On windows.

**Comment by ufomike** [comment]
@jef, thank you for the reply I tried your suggestion and got the player to load but unfortunately not the mp3 files. Can you please look into this matter since multiple users are experiencing the same issue? Does the sound file work with your .env files? Once again thank you for taking the time and looking into this.

**Comment by jef** [comment]
Unfortunately, this is user error 😢 I will look into the `INFO: Could not find files for the given pattern(s).` at some point in the future though.

```
> streetmerchant@ test:notification C:\Users\jef\Downloads\streetmerchant
> npm run build && node build/__test__/notification-test.js


> build
> tsc

ＳＴＲＥＥＴＭＥＲＣＨＡＮＴ
3.1.0

INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
INFO: Could not find files for the given pattern(s).
[6:50:56 PM] info :: ✔ sound player found: cmdmp3
```

I am able to play sound fine.
